src/main/scala/stdlib/Lists.scala

```diff
@@ -90,50 +90,93 @@ object Lists extends FlatSpec with Matchers with org.scalaexercises.definitions.
     b should equal(List(res0, res1, res2, res3))
   }
 
-  /** Lists have many useful utility methods:
-   */
-  def usefulMethodsLists(res0: Int, res1: List[Int], res2: List[Int], res3: List[Int]) {
+  /** Lists have many useful utility methods.
+    *
+    * `reverse` return a reversed copy of the list:
+    */
+  def reverseLists(res0: Int, res1: List[Int]) {
     val a = List(1, 3, 5, 7, 9)
 
     // get the length of the list
     a.length should equal(res0)
 
     // reverse the list
     a.reverse should equal(res1)
+  }
 
-    // map a function to double the numbers over the list
-    a.map { v ⇒
-      v * 2
-    } should equal(res2)
+  /** `filter` tests each element of the List by applying a function you supply:
+    */
+  def filterLists(res0: List[Int], res1: List[Int], res2: List[Int]) {
+    val a = List(1, 2, 3, 5, 7, 9)
 
     // filter any values divisible by 3 in the list
     a.filter { v ⇒
       v % 3 == 0
-    } should equal(res3)
+    } should equal(res0)
+
+    // Functions over lists can use _ as shorthand
+    a.filter {
+      _ % 2 == 0
+    } should equal(res1)
+
+    // Functions over lists can use () instead of {}
+    a.filter(_ % 2 != 0) should equal(res2)
   }
 
-  /** Functions over lists can use _ as shorthand:
-   */
-  def wildcardAsShorthandLists(res0: Int, res1: Int, res2: Int, res3: Int) {
-    val a = List(1, 2, 3)
+  /** `map` works by applying a function to each element in the list:
+    */
+  def mapLists(res0: List[Int], res1: List[Int], res2: List[Int]) {
+    val a = List(1, 3, 5, 7, 9)
 
+    // map a function to double the numbers over the list
+    a.map { v ⇒
+      v * 2
+    } should equal(res0)
+
+    // Functions over lists can use _ as shorthand
     a.map {
-      _ * 2
-    } should equal(List(res0, res1, res2))
+      _ + 2
+    } should equal(res1)
 
-    a.filter {
-      _ % 2 == 0
-    } should equal(List(res3))
+    // Functions over lists can use () instead of {}
+    a.map(_ * 2) should equal(res2)
   }
 
-  /** Functions over lists can use `()` instead of `{}`:
-   */
-  def functionsOverListsLists(res0: Int, res1: Int, res2: Int, res3: Int, res4: Int) {
-    val a = List(1, 2, 3)
-    a.map(_ * 2) should equal(List(res0, res1, res2))
-    a.filter(_ % 2 != 0) should equal(List(res3, res4))
+  /** `flatten` The flatten method takes a list of lists and flattens it out to a single list:
+    */
+  def flattenLists(res0: List[Int], res1: List[Int]) {
+    val a = List(List(1,2), List(3,4))
+    val b = List(1,2)
+
+    a.flatten should equal(res0)
+
+    b.map { v =>
```

Comment:
I think we don't necessarily have to duplicate the map.flatten = flatMap example here. What do you think?

Reply:
I think it will be better to have both the example here so that person can understand easily and co-related functions easily.